quic/s2n-quic/src/provider/connection_id.rs

```diff
@@ -111,11 +115,25 @@ pub mod default {
             Ok(self)
         }
 
+        /// Disables rotation of the connection Id used during the handshake
+        ///
+        /// By default the connection ID used during the the handshake
+        /// will be requested to be retired following confirmation of the handshake
+        /// completing. This reduces linkability between information exchanged
+        /// during and after the handshake.
+        pub fn with_handshake_connection_id_rotation_disabled(
+            mut self,
+        ) -> Result<Self, core::convert::Infallible> {
+            self.rotate_handshake_connection_id = false;
+            Ok(self)
+        }
```

Comment:
I'd prefer something like:

```suggestion
        pub fn with_handshake_connection_id_rotation(
            mut self,
            enabled: bool,
        ) -> Result<Self, core::convert::Infallible> {
            self.rotate_handshake_connection_id = enabled;
            Ok(self)
        }
```